server/lib/konnector_poller.coffee

```diff
@@ -18,187 +18,154 @@ month = 30 * day
-                        "#{nextUpdate.format(format)}"
-
-                        @prepareNextCheck konnector, interval
                         callback()
-                else
-                    callback()
-
-    handleTimeout: (konnector) ->
-
-        # if date is present in fieldValues
+                , (err) =>
+                    # Initialize every day to avoid to have lots of long timeout.
+                    @prepareNextCheck()
+                    if timeout?
+                        clearTimeout timeout
+                    timeout = setTimeout @start.bind(@), day
+                    cb() if cb?
+        else
+            @prepareNextCheck()
+            if timeout?
+                clearTimeout timeout
+            timeout = setTimeout @start.bind(@), day
+            cb() if cb?
+
+    findNextUpdate: (konnector) ->
+        importInterval = periods[konnector.importInterval]
+        now = moment()
+        lastImport = moment(konnector.lastImport)
+        lastAutoImport = moment(konnector.lastAutoImport)
+
+        # if we missed an importation cycle
+        if (now.valueOf() - lastAutoImport.valueOf()) > importInterval
+            log.debug "#{konnector.slug} missed an importation cycle"
+            # We import now
+            importer konnector
+
+            # calculate the supposed last Auto-import
+            importTime = lastAutoImport.add importInterval, 'ms'
+            # calculate the time elapsed
+            while importTime.valueOf() < now.valueOf()
+                importTime = importTime.add importInterval, 'ms'
+
+            log.debug "#{konnector.slug} | Next update : " +
+            "#{importTime.format(format)}"
+            return importTime
+            callback()
+
+        # if we didn't missed an import interval
+        else
+            if lastAutoImport.valueOf() > now.valueOf()
+                # possible if user precises a start date for auto import
+                # TODOS : manage start date in other field than lastAutoImport
+                nextUpdate = lastAutoImport
+            else
+                nextUpdate = lastAutoImport.add importInterval, 'ms'
+            log.debug "#{konnector.slug} | Next update : " +
+            "#{nextUpdate.format(format)}"
+            return nextUpdate
+
+    handleTimeout: (konnector, cb=null) ->
+        # Update timeouts and nextUpdates for this new/modified konnector
+
+        # If date is present in fieldValues
         startDate = konnector.fieldValues.date if konnector.fieldValues.date?
         # Retrive current Autoimport value in database
         Konnector.find konnector.id, (err, savedKonnector) =>
             savedKonnector.injectEncryptedFields()
-
             currentInterval = savedKonnector.importInterval
 
-            # If the importInterval has changed
+            # If the importInterval has changed or it's a new
             if konnector.importInterval isnt currentInterval
```

Comment:
We may change the way handleTimeout works, or change how the save/import (in the controller) is done, because based on https://github.com/cozy-labs/konnectors/blob/master/server/controllers/konnectors.coffee#L39, the code cannot enter in that condition. I feared I broke that...confirmed.